### DISCUSSION TITLE
Relations field "sources" option to only show entries with an URL

### DISCUSSION THREAD

**Discussion by FrDH** [craftcms/cms#12018] [Ideas] [answered]
Somewhat related to [this LinkIt issue](https://github.com/presseddigital/linkit/issues/105), it would be useful if there was an "All with URL" option for the "sources" setting in a relations field.
This way, when we use a relations field to pick entries to link to, the user isn't able to pick entries that don't have an URL.
Hope this makes sense :)

**Comment by brandonkelly** [comment, marked as answer]
This is possible in Craft 4 thanks to relation fields’ new “Selectable [element type] Condition” settings:

<img width="595" alt="The “Selectable Entries Condition” setting for an Entries field, with a “Has URL” rule selected" src="https://user-images.githubusercontent.com/47792/194439013-9ec77d22-8261-4664-a043-0125bcb484da.png">
